### PR TITLE
Refactoring downloads controller

### PR DESCRIPTION
**Pull request by cgalarza**
This is in preparation for adding a batch_imports download endpoint. For which we have to provide a download to the original file and a file without the successful rows.

### CODE REVIEW

**Comment by cgalarza** [comment]
@elohanlon I tried making the `download` method take all the parameters needed, but the total number of parameters was 6 and rubocop was not happy. 

Another thing we could do at the end of each action method is call `set_download_header` and `download` separately.

**Comment by elohanlon** [comment]
@cgalarza This seems like a case where the benefits of overriding the rubocop complaint (via `# rubocop:disable` comment) may outweigh the downsides of having to set a bunch of instance variables.  The instance variable method is a little more error prone. Would you be open to doing that?

**Comment by elohanlon** [comment]
But also, calling `set_download_header` and `download` separately is a good idea.  If that keeps rubocop happy, it's definitely still better than relying on the setting of a bunch of instance variables.

**Comment by cgalarza** [comment]
Okay, sounds good, i'll go for calling the two methods separately.

**Comment by cgalarza** [comment]
Just pushed an update.

File: app/controllers/api/v1/downloads_controller.rb [hunk 1]
```diff
@@ -9,79 +9,64 @@ class DownloadsController < ApplicationApiController
...
-          while (chunk = io.read(DOWNLOAD_BUFFER_BYTE_SIZE))
-            response.stream.write(chunk)
-          end
-        ensure
-          response.stream.close
-        end
+      # GET /downloads/digital_object/:id/:resource_name
+      def digital_object
+        id = digital_object_params[:id]
+        subresource = digital_object_params[:resource_name]
```
Comment (elohanlon) [inline app/controllers/api/v1/downloads_controller.rb]:
```suggestion
        resource_name = digital_object_params[:resource_name]
```

File: app/controllers/api/v1/downloads_controller.rb [hunk 2]
```diff
@@ -9,79 +9,64 @@ class DownloadsController < ApplicationApiController
...
+
+        digital_object = DigitalObject::Base.find(id)
+        authorize! :read, digital_object
+        resource = digital_object.resources[subresource]
```
Comment (elohanlon) [inline app/controllers/api/v1/downloads_controller.rb]:
```suggestion
        resource = digital_object.resources[resource_name]
```

**Comment by elohanlon** [comment]
Looks great!

**Pull request merged**